src/overlays/actors/ovl_Obj_Yado/z_obj_yado.c

```diff
@@ -28,22 +27,50 @@ const ActorInit Obj_Yado_InitVars = {
     (ActorFunc)ObjYado_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80C16420[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+AnimatedMaterial* D_80C16470;
 
-extern InitChainEntry D_80C16420[];
+extern Gfx D_06000320[];
+extern Gfx D_06000430[];
+extern AnimatedMaterial D_060012E8;
 
-extern UNK_TYPE D_06000430;
-extern UNK_TYPE D_060012E8;
+void ObjYado_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYado* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yado/ObjYado_Init.s")
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    D_80C16470 = (AnimatedMaterial*)Lib_SegmentedToVirtual(&D_060012E8);
+    this->colorFlag = gSaveContext.isNight;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yado/ObjYado_Destroy.s")
+void ObjYado_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yado/ObjYado_Update.s")
+void ObjYado_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjYado* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Yado/ObjYado_Draw.s")
+    this->colorFlag = gSaveContext.isNight;
+}
+
+void ObjYado_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjYado* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (this->colorFlag != 0) {
+        gSPSegment(POLY_XLU_DISP++, 0x09, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 95, 95, 70, 155));
+        gSPSegment(POLY_OPA_DISP++, 0x0A, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 0, 40, 40, 255));
+    } else {
+        gSPSegment(POLY_XLU_DISP++, 0x09, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 255, 255, 215, 110));
+        gSPSegment(POLY_OPA_DISP++, 0x0A, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 255, 255, 215, 255));
+    }
```

Comment:
```suggestion
    if (this->isNight) {
        gSPSegment(POLY_XLU_DISP++, 0x09, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 95, 95, 70, 155));
        gSPSegment(POLY_OPA_DISP++, 0x0A, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 0, 40, 40, 255));
    } else {
        gSPSegment(POLY_XLU_DISP++, 0x09, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 255, 255, 215, 110));
        gSPSegment(POLY_OPA_DISP++, 0x0A, Gfx_PrimColor(globalCtx->state.gfxCtx, 128, 255, 255, 215, 255));
    }
```
This is more personal preference, but keeping the name of the flag to what the flag is representing seems clearer to me. `this->colorFlag = gSaveContext.isNight;` looks a little strange. But that's not a strong opinion. I'll leave it up to you.

In either case, I think this is a bool and the `!= 0` can go